src/middleware/node/get-payload.ts

```diff
@@ -11,20 +11,13 @@ import AggregateError from "aggregate-error";
 // }
 type IncomingMessage = any;
 
-export function getPayload(request: IncomingMessage): Promise<string> {
-  // If request.body already exists we can stop here
-  // See https://github.com/octokit/webhooks.js/pull/23
-
-  if (request.body) return Promise.resolve(request.body);
-
+export function getPayload(request: IncomingMessage): Promise<Buffer> {
   return new Promise((resolve, reject) => {
-    let data = "";
-
-    request.setEncoding("utf8");
+    let data: Buffer[] = [];
 
     // istanbul ignore next
     request.on("error", (error: Error) => reject(new AggregateError([error])));
-    request.on("data", (chunk: string) => (data += chunk));
-    request.on("end", () => resolve(data));
+    request.on("data", (chunk: Buffer) => data.push(chunk));
+    request.on("end", () => resolve(Buffer.concat(data)));
```

Comment:
```suggestion
    request.on("data", data.push.bind(data));
    request.on("end", setImmediate(resolve, Buffer.concat(data)));
}
```

Reply:
@kibertoad
It seems that the setImmediate improves the performance. Seems like without it the event loop is clogging? Maybe it helps to relax the verification later, which is doing a createHmac and is IO heavy and thus hurting the performance of the event loop?

Reply:
Buffer concatenation happens immediately in this code, so the only thing deferred is the promise resolution.

ChatGPT has this to say: 
![image](https://github.com/octokit/webhooks.js/assets/1847934/ac4386f1-3b2f-4c3d-b5d4-cca9a0548322)

Reply:
Oh right. If I do () => setImmediate it works.